src/trio/_tests/type_tests/raisesgroup.py

```diff
@@ -84,9 +79,9 @@ def check_filenotfound(exc: FileNotFoundError) -> bool:
     Matcher(exception_type=ValueError)
     Matcher(match="regex")
     Matcher(check=check_exc)
-    Matcher(check=check_filenotfound)  # type: ignore
     Matcher(ValueError, match="regex")
     Matcher(FileNotFoundError, check=check_filenotfound)
+    Matcher(check=check_filenotfound)  # type: ignore # not narrowed
```

Comment:
What does "not narrowed" mean in this situation? Does it mean not ignoring specific type (such as `# type: ignore[misc]`)? If so why?

Reply:
I could clarify the comment, though there is a comment a few lines up not seen in the preview that explains it. I simply moved the test to be next to the other `check=check_filenotfound` line.
It fails because `check_filenotfound` has its arg typed as `FileNotFoundError` but the `Matcher` doesn't narrow the type from `Exception` when there's no type passed.

Specific ignore types are not used in type tests due to how pyright/mypy differs. Even if you ignore a specific type with `# type: ignore` pyright will ignore all warnings on the line, because it does not have the same mapping of warning names. I suppose it can still give some information and give warning if mypy warning changes, but it's also hella misleading wrt pyright.

Reply:
Thanks for clarifying, yea I didn't see the comment in the changes preview